cmake/extra_warnings.cmake

```diff
@@ -0,0 +1,118 @@
+# Enable some useful warnings
+ADD_DEFINITIONS(
+   -Wall
+   -Wextra
+   -Wmissing-declarations
+   -Wmissing-noreturn
+   -Wpointer-arith
+   -Wcast-align
+   -Wwrite-strings
+   -Wformat-nonliteral
+   -Wformat-security
+   -Wswitch-enum
+   -Winit-self
+   -Wmissing-include-dirs
+   -Wundef
+   -Wmissing-format-attribute
+   -Wunused
+   -Wuninitialized
+   -Wunused-value
+   -pedantic
+   -Wnonnull
+   -Wsequence-point
+   #-Wsystem-headers
+   -Wsizeof-pointer-memaccess
+   -Wvarargs
+   -Wmaybe-uninitialized
+   -Wunused-local-typedefs
+   -Wno-unused-function
+   -Wno-attributes
+   -Wno-missing-include-dirs
+   -Wno-unknown-pragmas
+
+   # Too noisy
+   -Wno-pedantic
+)
+
+#Add more warnings for compilers that support it. I used this command:
+#curl https://gcc.gnu.org/onlinedocs/gcc-4.9.2/gcc/Warning-Options.html | \
+#grep -E "^[\t ]+<br><dt><code>-W[a-zA-Z=-]*" -o | grep -E "\-W[a-zA-Z=-]*" -o >
+#cat /tmp/48 /tmp/49 | sort | uniq -u
```

Comment:
A part of the command line is missing...? (The trailing `>` is either a broken file redirection or an indication that the line did not fit onto your screen)